### PR TITLE
refactor: convert field requirements to validators

### PR DESCRIPTION
**Pull request by ozgurkececioglu**
[FX-NNNN]

### Description

Describe the changes and motivations for the pull request.

### How to test

- FIXME: Add the steps describing how to verify your changes

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| Insert screenshots or screen recordings | Insert screenshots or screen recordings |

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

### CODE REVIEW

File: packages/picasso-forms/src/Field/Field.tsx [hunk 1]
```diff
@@ -170,7 +186,7 @@ const Field = <
 
   return (
     <PicassoForm.Field
-      error={error}
+      error={showRequirements ? { error, validate } : error}
```
Comment (ozgurkececioglu) [inline packages/picasso-forms/src/Field/Field.tsx:189]: The logic for the `error` prop is still a mystery. We need to pass an array of error messages for the success state. This means that I need to call each requirement twice: one for actual validation, and one for getting the message if it succeed.

File: packages/picasso-forms/src/Form/story/index.jsx [hunk 2]
```diff
@@ -137,96 +137,96 @@ Form is a wrapper for 'react-final-form' Form component. It also
...
-  .addExample(
-    'Form/story/ParseInput.example.tsx',
-    {
-      title: 'Change form input value',
-      description: `
-      When you use picasso-forms your form input components are no longer
-      completely controlled and they are controlled by final-form, which
-      gives you the opportunity to rely on it with displaying errors,
-      validations, etc.
+  //   .addExample(
```
Comment (ozgurkececioglu) [inline packages/picasso-forms/src/Form/story/index.jsx:140]: Just ignored everything to suppress the warnings

File: packages/picasso-forms/src/PasswordInput/PasswordInput.tsx [hunk 3]
```diff
@@ -1,106 +1,33 @@
...
-      setShowContent(false)
-    }, ANIMATION_TIMEOUT)
-  }, [])
+const passwordValidators = composeRequirementsValidators([
+  validators.atLeastEightCharacters,
+  validators.atLeastOneNumber,
+  validators.atLeastOneUpperCaseCharacter,
+  validators.atLeastOneLowerCaseCharacter,
+  validators.atLeastOneSpecialCharacter
+])
```
Comment (denieler) [inline packages/picasso-forms/src/PasswordInput/PasswordInput.tsx:22]: this looks nice 👍

File: packages/picasso-forms/src/PasswordInput/PasswordInput.tsx [hunk 4]
```diff
@@ -1,106 +1,33 @@
...
   return (
     <InputField<PasswordInputProps>
       {...rest}
-      validate={validationsObject}
-      renderFieldRequirements={
-        !hideRequirements ? renderFieldRequirements : undefined
-      }
+      validate={passwordValidators}
+      requirementsDescription='Please make sure that your password contains:'
+      showRequirements
```
Comment (denieler) [inline packages/picasso-forms/src/PasswordInput/PasswordInput.tsx:30]: do we really need those 2 last props?

**Comment by ozgurkececioglu** [comment]
I'm closing this to get back to it later at a better time.

**Pull request closed without merging**